File: src/my_ude_py/scripts/speedometer.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from std_msgs.msg import Float32

WHEELRADIO_DEFFAULT_=1.0;

class SpeedometerNode(Node): 
    def __init__(self):
        super().__init__("speedometer") 
        self.declare_parameter("wheel_radio",WHEELRADIO_DEFFAULT_)
        self.suscreiber_=self.create_subscription(Float32,"rpm",self.call_rpm,10)
        self.publisher_=self.create_publisher(Float32,"velocity",10)
        self.get_logger().info("Speedometer has been started")

    def call_rpm(self, msg):
        self.get_logger().info("Los RPM son de :" + str(msg.data))
        message=Float32()
        self.wheelradio_= self.get_parameter("wheel_radio").get_parameter_value().double_value
        message.data=self.wheelradio_*msg.data
        self.publisher_.publish(message)

def main(args=None):
    rclpy.init(args=args)
    node = SpeedometerNode() 
    rclpy.spin(node)
    rclpy.shutdown()


if __name__ == "__main__":
    main()

File: src/my_ude_cpp/src/speedometer.cpp
#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/float32.hpp"

using std::placeholders::_1;

const double wheelradio_default = 1.0;

class SpeedometerNode : public rclcpp::Node 
{
public:
    SpeedometerNode() : Node("speedometer")
    {
        this->declare_parameter("wheel_radio",wheelradio_default);
        publisher_=this->create_publisher<std_msgs::msg::Float32>("velocity",10);
        subscriber_ = this-> create_subscription<std_msgs::msg::Float32>("rpm",10,
                std::bind(&SpeedometerNode::callRpm, this, _1));
        RCLCPP_INFO(this->get_logger(), "Speedometer has been started.");
    }

private:

    void callRpm(const std_msgs::msg::Float32::SharedPtr msg){
        RCLCPP_INFO(this->get_logger(), "Los RPM son de: %f", msg ->data);
        auto message=std_msgs::msg::Float32();
        wheelradio_=this->get_parameter("wheel_radio").as_double();
        message.data= msg->data*wheelradio_;
        publisher_->publish(message);
    }

    float wheelradio_;

    rclcpp::Subscription<std_msgs::msg::Float32>::SharedPtr subscriber_;
    rclcpp::Publisher<std_msgs::msg::Float32>::SharedPtr publisher_;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<SpeedometerNode>(); 
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}